honeybee_energy/material/_base.py

```diff
@@ -0,0 +1,91 @@
+# coding=utf-8
+"""Base energy material."""
+from __future__ import division
+
+from honeybee.typing import valid_ep_string
+
+
+class _EnergyMaterialBase(object):
+    """Base energy material.
+
+    Properties:
+        name
+    """
+    _name_descr = 'material name'
+    _is_window_material = None
+
+    def __init__(self, name):
+        """Initialize energy material base.
+
+        Args:
+            name: Text string for material name. Must be <= 100 characters.
+                Can include spaces but special characters will be stripped out.
+        """
+        self.name = name
+
+    @property
+    def name(self):
+        """Get or set the text string for material name."""
+        return self._name
+
+    @name.setter
+    def name(self, name):
+        self._name = valid_ep_string(name, self._name_descr)
+
+    @property
+    def is_window_material(self):
+        """Boolean to note whether the material can be used for window surfaces."""
+        return self._is_window_material
+
+    def duplicate(self):
+        """Get a copy of this construction."""
+        return self.__copy__()
+
+    def __copy__(self):
+        return self.__class__(self.name)
+
+    def ToString(self):
+        """Overwrite .NET ToString."""
+        return self.__repr__()
+
+    def __repr__(self):
+        return 'Base Energy Material:\n{}'.format(self.name)
+
+
+class _EnergyMaterialOpaqueBase(_EnergyMaterialBase):
+    """Base energy material for all opaque material types."""
+    _is_window_material = False
+    _rough_types = ('VeryRough', 'Rough', 'MediumRough',
```

Comment:
This looks like a list of constant values to me. You may want to use `ROUGHTYPES` instead.

Reply:
Ah. Good point. I forgot that convention. Ok I will use that throughout the library.